src/components/LogoutTile.jsx

```diff
@@ -1,20 +1,27 @@
 import React, { useCallback } from 'react'
-import Icon from 'cozy-ui/transpiled/react/Icon'
-import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+
 import { useClient } from 'cozy-client'
+import { useWebviewIntent } from 'cozy-intent'
+import { isFlagshipApp } from 'cozy-device-helper'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+import Icon from 'cozy-ui/transpiled/react/Icon'
 import useBreakpoints from 'cozy-ui/transpiled/react/hooks/useBreakpoints'
 import SquareAppIcon from 'cozy-ui/transpiled/react/SquareAppIcon'
-
 import LogoutLargeIcon from 'cozy-ui/transpiled/react/Icons/LogoutLarge'
 
 const LogoutTile = () => {
   const { t } = useI18n()
   const client = useClient()
+  const webviewIntent = useWebviewIntent()
 
   const logout = useCallback(async () => {
     await client.logout()
-    window.location.reload()
-  }, [client])
+
+    return isFlagshipApp() && webviewIntent
+      ? webviewIntent.call('logout')
+      : window.location.reload()
+  }, [client, webviewIntent])
```

Comment:
nit: in my opinion, we should add one test on that feature (or an issue to add it) 🙏

Reply:
Not in my opinion. We should not have this kind of stuff here. We should not test if we are in flagship or else. We sould only rely on one method. I'll create an issue later on this point.